coderd/database/pubsub/pubsub.go

```diff
@@ -528,26 +566,45 @@ func (p *PGPubsub) Collect(metrics chan<- prometheus.Metric) {
 	p.qMu.Unlock()
 	metrics <- prometheus.MustNewConstMetric(currentSubscribersDesc, prometheus.GaugeValue, float64(subs))
 	metrics <- prometheus.MustNewConstMetric(currentEventsDesc, prometheus.GaugeValue, float64(events))
+
+	// additional metrics
+	latency := p.latencyMeasurer.LastMeasurement()
+	if latency == nil {
+		p.logger.Debug(context.Background(), "latency measurement not completed yet")
+		return
+	}
+
+	metrics <- prometheus.MustNewConstMetric(pubsubLatencyMeasureCountDesc, prometheus.CounterValue, float64(p.latencyMeasurer.MeasurementCount()))
+	if latency.Err != nil {
+		p.logger.Warn(context.Background(), "failed to measure latency", slog.Error(latency.Err))
+		metrics <- prometheus.MustNewConstMetric(pubsubLatencyMeasureErrDesc, prometheus.CounterValue, float64(p.latencyErrCounter.Add(1)))
+		return
+	}
+	metrics <- prometheus.MustNewConstMetric(pubsubSendLatencyDesc, prometheus.GaugeValue, latency.Send.Seconds())
+	metrics <- prometheus.MustNewConstMetric(pubsubRecvLatencyDesc, prometheus.GaugeValue, latency.Recv.Seconds())
 }
 
 // New creates a new Pubsub implementation using a PostgreSQL connection.
-func New(startCtx context.Context, logger slog.Logger, database *sql.DB, connectURL string) (*PGPubsub, error) {
+func New(startCtx context.Context, logger slog.Logger, database *sql.DB, connectURL string, latencyMeasureInterval time.Duration) (*PGPubsub, error) {
 	p := newWithoutListener(logger, database)
 	if err := p.startListener(startCtx, connectURL); err != nil {
 		return nil, err
 	}
 	go p.listen()
 	logger.Info(startCtx, "pubsub has started")
+	go p.latencyMeasurer.MeasureAsync(startCtx, p, latencyMeasureInterval)
```

Comment:
`startCtx` just represents how long we're willing to wait for the PubSub to start up, it's not the full lifetime of the PubSub.  The PubSub is explicitly closed, and so it probably makes the most sense for the async measurement to also be explicitly closed.

Reply:
Thanks!